modules/integration/tests-integration/tests-patches/src/test/java/org/wso2/carbon/esb/json/ESBJAVA_3698_MessageBuildingWithDifferentPayloadAndContentTypeTestCase.java

```diff
@@ -0,0 +1,93 @@
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.base.CarbonBaseUtils;
+import org.wso2.carbon.esb.util.FileUtil;
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+
+public class ESBJAVA_3698_MessageBuildingWithDifferentPayloadAndContentTypeTestCase extends
+                                                                                   ESBIntegrationTest {
+    private final DefaultHttpClient httpClient = new DefaultHttpClient();
+    final String carbonLogFile = CarbonBaseUtils.getCarbonHome() +
+                                 "/repository/logs/wso2carbon.log";
+    private ServerConfigurationManager serverConfigurationManager;
+
+    @BeforeClass(alwaysRun = true)
+    public void setEnvironment() throws Exception {
+        super.init();
+
+        serverConfigurationManager = new ServerConfigurationManager(context);
+        loadESBConfigurationFromClasspath("artifacts" + File.separator + "ESB" + File.separator +
+                                          "json" + File.separator + "StockQuoteAPI.xml");
+
+        final String log4jSourceFile =
+                                       System.getProperty("framework.resource.location") +
+                                               "/artifacts/ESB/json/log4j.properties";
+        final String log4jDestinationFile =
+                                            CarbonBaseUtils.getCarbonHome() +
+                                                    "/repository/conf/log4j.properties";
+
+        serverConfigurationManager.applyConfiguration(new File(log4jSourceFile),
+                                                      new File(log4jDestinationFile));
+        super.init();
+    }
+
+    @Test(groups = { "wso2.esb" }, description = "Check for Axis Fault when xml payload is sent with application/json content type")
+    public void testAxisFaultWithXmlPayloadAndJSONContentType() throws ClientProtocolException,
+                                                               IOException, InterruptedException {
+        final Map<String, String> headers = new HashMap<String, String>();
+
+        final HttpPost post = new HttpPost("http://localhost:8280/stockquote/test");
+        post.addHeader("Content-Type", "application/json");
+        post.addHeader("SOAPAction", "urn:getQuote");
+        StringEntity se = new StringEntity(getPayload());
+
+        post.setEntity(se);
+        HttpResponse response = httpClient.execute(post);
+
+        Thread.sleep(2000);
```

Comment:
Can we just throw interrupted Exception best way to handle is Thread.currentThread().interrupt();